### PR TITLE
Fix maven build ${version} deprecation warning

### PR DESCRIPTION
**Pull request by cjcjameson**
Signed-off-by: C.J. Jameson <cjcjameson@gmail.com>

##  opensearch-security pull request intake form

 1. __Category:__ 
 
 maintenance
 
 2. __Github Issue # or road-map entry, if available:__

n/a

 3. __Description of changes:__

`mvn package` warns...

[WARNING] Some problems were encountered while building the effective model for org.opensearch:opensearch-security:jar:1.0.0.0-rc1
[WARNING] The expression ${version} is deprecated. Please use ${project.version} instead.
[WARNING] The expression ${version} is deprecated. Please use ${project.version} instead.
[WARNING]
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING]
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.

 4. __Why these changes are required?__ 

make it easier to upgrade pom.xml dependencies

 5. __What is the old behavior before changes and new behavior after changes?__ (_Please add any example/logs/screen-shot if available_)

 6. __Testing done:__ (_Please provide details of testing done: Unit testing, integration testing and manual testing_)
 
 
 
 7. __TO-DOs, if any:__ (_Please describe pending items and provide Github issues# for each of them_)

 8. __Is it backport from main branch?__ (_If yes, please add backport PR # and commits #_)

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vrozov** [comment]
Please rebase and resolve conflicts

**Comment by cjcjameson** [comment]
@vrozov rebased

**Review by vrozov — APPROVED** [review]
LGTM

**Review by cliu123 — DISMISSED** [review]

**Comment by cliu123** [comment]
@cjcjameson Would you please sign the commit and push it one more time?

**Comment by cjcjameson** [comment]
@vrozov @cliu123 signed and rebased, thanks!

**Review by cliu123 — DISMISSED** [review]

**Review by vrozov — APPROVED** [review]

**Comment by cjcjameson** [comment]
@vrozov all set for you to merge

File: build.gradle [hunk 1]
```diff
@@ -88,10 +88,10 @@ ospackage {
 buildRpm {
     arch = 'NOARCH'
     addParentDirs = false
-    archiveName "${packageName}-${version}.rpm"
+    archiveName "${packageName}-${project.version}.rpm"
```
Comment (vrozov) [inline build.gradle:91]: `version` for gradle is defined on line 59.

Comment (cjcjameson) [reply build.gradle:91]: @vrozov can you make the fix you recommend and merge it? I don't understand the specific feedback you're giving, and the overall PR is so small; I am feeling frustrated with all the rebasing, signing, two reviewers, etc.

Comment (vrozov) [reply build.gradle:91]: @cjcjameson It is not necessary to modify `build.gradle`. It is not part of the maven build and is not a subject of the maven deprecation warning that this PR is fixing.

Comment (cjcjameson) [reply build.gradle]: ok, gotcha. removed

**Comment by cjcjameson** [comment]
Re-running CI system. Failure seems unrelated.

```
Error: 7-19T20:11:09,385][ERROR][org.opensearch.security.test.helper.file.FileHelper] Failed to load dlsfls/nodes_dn.yml
Error: 7-19T20:11:09,386][ERROR][org.opensearch.security.test.helper.file.FileHelper] Failed to load dlsfls/whitelist.yml
```

**Review by vrozov — APPROVED** [review]
LGTM, thank you for contribution.

**Review by cliu123 — APPROVED** [review]

**Comment by cjcjameson** [comment]
@cliu123 can you be the one to merge? I cannot.

**Pull request merged**